### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by HStakhiv**

### CODE REVIEW

File: db/queries.py [hunk 1]
```diff
@@ -1,46 +1,75 @@
...
 def get_messages_contain_authors_first_name() -> list[Message]:
-    pass
+    return list(Message.objects.filter(text__contains=F("user__first_name")))
 
 
 def get_users_who_sent_messages_starts_with_m_or_a() -> list[User]:
-    pass
+    return list(
+        User.objects.filter(
+            Q(message__text__startswith="a") | Q(message__text__startswith="m")
```
Comment (Yaroslav-Stetsko) [inline db/queries.py:33]: Use case-insensitive containment test.

File: db/queries.py [hunk 2]
```diff
@@ -1,46 +1,75 @@
-from db.models import Message, User, Chat
 from django.db.models import Q, Count, F
 
+from db.models import Message, User, Chat
+
 
 def get_messages_that_contain_word(word: str) -> list[Message]:
-    pass
+    return list(Message.objects.filter(text__contains=word))
```
Comment (Yaroslav-Stetsko) [inline db/queries.py:7]: Use case-insensitive containment test.

Comment (HStakhiv) [reply db/queries.py:7]: You're right.  I have to read the paragraphs to the end

**Review by Yaroslav-Stetsko — APPROVED** [review]
Looks great

File: db/queries.py [hunk 3]
```diff
@@ -1,46 +1,76 @@
...
+    return list(
+        User.objects.filter(message__sent__year="2015").values_list(
+            "first_name", "last_name"
+        )
+    )
 
 
 def get_actual_chats() -> list[Chat]:
-    pass
+    return list(Chat.objects.filter(message__sent__year__gt="2020").distinct())
```
Comment (Nattalli) [inline db/queries.py:23]: does `.distinct()` needed here?

**Review by Nattalli — CHANGES_REQUESTED** [review]

File: db/queries.py (hunk 3 above)
Comment (HStakhiv) [reply db/queries.py:23]: No

**Review by Nattalli — APPROVED** [review]

**Pull request closed without merging**